sway-fmt-v2/src/items/item_enum.rs

```diff
@@ -4,10 +4,77 @@ use crate::{
     utils::bracket::CurlyDelimiter,
 };
 use sway_parse::ItemEnum;
+use sway_types::Spanned;
 
 impl Format for ItemEnum {
-    fn format(&self, _formatter: &mut Formatter) -> FormattedCode {
-        todo!()
+    fn format(&self, formatter: &mut Formatter) -> FormattedCode {
+        // TODO: creating this formatted_code with String::new() will likely cause lots of
+        // reallocations maybe we can explore how we can do this, starting with with_capacity may help.
+        let mut formatted_code = String::new();
+        let enum_variant_align_threshold = formatter.config.structures.enum_variant_align_threshold;
+
+        if let Some(visibility_token) = &self.visibility {
+            formatted_code.push_str(visibility_token.span().as_str());
+            formatted_code.push(' ');
+        }
+
+        // Add enum token
+        formatted_code.push_str(self.enum_token.span().as_str());
+        formatted_code.push(' ');
+
+        // Add name of the enum.
+        formatted_code.push_str(self.name.as_str());
+        ItemEnum::handle_open_brace(&mut formatted_code, formatter);
+
+        let type_fields = &self.fields.clone().into_inner().value_separator_pairs;
+
+        // In first iteration we are going to be collecting the lengths of the enum variants.
+        let variant_length: Vec<usize> = type_fields
+            .iter()
+            .map(|variant| variant.0.name.as_str().len())
+            .collect();
+
+        // Find the maximum length in the variant_length vector that is still smaller than enum_variant_align_threshold.
+        let mut max_valid_variant_length = 0;
+
+        variant_length.iter().for_each({
+            |length| {
+                if *length > max_valid_variant_length && *length < enum_variant_align_threshold {
+                    max_valid_variant_length = *length;
+                }
+            }
+        });
```

Comment:
I don't think the extra nested braces are necessary here, e.g. should be able to do:

```rust
variant_length.iter().for_each(|length| {
    if *length > max_valid_variant_length && *length < enum_variant_align_threshold {
        max_valid_variant_length = *length;
    }
});
```